front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/ApplicationsController.groovy

```diff
@@ -97,31 +110,36 @@ public class ApplicationsController {
     return app
   }
 
+  @PreAuthorize("hasPermission(#app.name, 'APPLICATION', 'CREATE')")
   @ApiOperation(value = "", notes = "Create an application within a specific `account`")
   @RequestMapping(method = RequestMethod.POST, value = "/name/{application:.+}")
   Application post(@RequestBody final Application app) {
     return getApplication().initialize(app).withName(app.getName()).save()
   }
 
+  @PreAuthorize("hasPermission(#application, 'APPLICATION', 'WRITE')")
   @ApiOperation(value = "", notes = "Delete an application from a specific `account`")
   @RequestMapping(method = RequestMethod.DELETE, value = "/name/{application:.+}")
   void delete(@PathVariable String application, HttpServletResponse response) {
     getApplication().initialize(new Application().withName(application)).delete()
     response.setStatus(HttpStatus.ACCEPTED.value())
   }
 
+  @PreAuthorize("hasPermission(#application, 'APPLICATION', 'READ')")
   @ApiOperation(value = "", notes = "Fetch a single application by name within a specific `account`")
   @RequestMapping(method = RequestMethod.GET, value = "/name/{application:.+}")
   Application getByName(@PathVariable final String application) {
     return getApplication().findByName(application)
   }
 
+  @PreAuthorize("hasPermission(#application, 'APPLICATION', 'READ')")
   @RequestMapping(value = '{application:.+}/history', method = RequestMethod.GET)
   Collection<Application> getHistory(@PathVariable String application,
                                      @RequestParam(value = "limit", defaultValue = "20") int limit) {
     return applicationDAO.getApplicationHistory(application, limit)
   }
 
+  @PreFilter("hasPermission(filterTarget.name, 'APPLICATION', 'WRITE')")
```

Comment:
should this be a PreAuthorize?

if I request to update a bunch of applications and only have permission to some of them, I'd expect the request to be rejected rather than partially applied to the apps I'm allowed